react/features/stream-effects/audio-mixer/AudioMixerEffect.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+
+import JitsiMeetJS from '../../base/lib-jitsi-meet';
+import { MEDIA_TYPE } from '../../base/media';
+import logger from '../../base/redux/logger';
+
+/**
+ * Class Implementing the effect interface expected by a JitsiLocalTrack.
+ * The AudioMixerEffect, as the name implies, mixes two JitsiLocalTracks containing a audio track. First track is
+ * provided at the moment of creation, second is provided through the effect interface.
+ */
+export class AudioMixerEffect {
+    /**
+     * JitsiLocalTrack that is going to be mixed into the track that uses this effect.
+     */
+    _mixAudio: Object;
+
+    /**
+     * lib-jitsi-meet AudioMixer.
+     */
+    _audioMixer: Object;
+
+    /**
+     * Creates AudioMixerEffect.
+     *
+     * @param {JitsiLocalTrack} mixAudio - JitsiLocalTrack which will be mixed with the original track.
+     */
+    constructor(mixAudio: Object) {
+        if (mixAudio.getType() !== MEDIA_TYPE.AUDIO) {
+            logger.warn('AudioMixerEffect only supports audio JitsiLocalTracks; effect will not work!');
+        }
+
+        this._mixAudio = mixAudio;
+    }
+
+    /**
+     * Checks if the JitsiLocalTrack supports this effect.
+     *
+     * @param {JitsiLocalTrack} sourceLocalTrack - Track to which the effect will be applied.
+     * @returns {boolean} - Returns true if this effect can run on the specified track, false otherwise.
+     */
+    isEnabled(sourceLocalTrack: Object) {
+        // Both JitsiLocalTracks need to be audio i.e. contain an audio MediaStreamTrack
+        return sourceLocalTrack.isAudioTrack() && this._mixAudio.isAudioTrack();
+    }
+
+    /**
+     * Effect interface called by source JitsiLocalTrack, At this point a WebAudio ChannelMergerNode is created
+     * and and the two associated MediaStreams are connected to it; the resulting mixed MediaStream is returned.
+     *
+     * @param {MediaStream} audioStream - Audio stream which will be mixed with _mixAudio.
+     * @returns {MediaStream} - MediaStream containing both audio tracks mixed together.
+     */
+    startEffect(audioStream: MediaStream) {
+        this._audioMixer = JitsiMeetJS.createAudioMixer();
+        this._audioMixer.addJitsiLocalTrack(this._mixAudio);
```

Comment:
this looks a bit odd, we have 2 apis to basically do the same. at the mixer level I'd say we only talk about streams WDYT?

Reply:
Yes it does feel a bit odd, the reason I added it is because during an older PR someone told me that we shouldn't access track.stream from jitis-meet as it would break the concept behind lib-jitsi-meet i.e. not accessing low level API such as MediaStream directly.

Reply:
Even though we access a MediaStream object thorough the effect interface ^^.
So I added it here even though I felt the same way you do at this point, so if you want I'd be more than happy to take it out.